### PR TITLE
add more checks

### PR DESCRIPTION
**Pull request by cbouss**
Add more checks:
has_run_test_and_commands
missing_pip_check
missing_wheel
missing_python
remove_python_pinning
gui_app

### CODE REVIEW

**Review by marcoesters — COMMENTED** [review]
Could there also be github packages that require `pip check`? As far as I know, they should be added for all "pure python" packages, so I don't see why this is domain-specific.

File: anaconda_linter/lint/check_build_help.py [hunk 1]
```diff
@@ -195,7 +195,157 @@ def check_source(self, source, section):
...
+          - pip check
+    """
+
+    def check_recipe(self, recipe):
+
+        # is it a pypi package?
+        pypi_source = False
+        source = recipe.get("source", None)
+        if isinstance(source, dict):
+            pypi_source = "pypi.io" in source.get("url", "")
```
Comment (skupr-anaconda) [inline anaconda_linter/lint/check_build_help.py:241]: `pypi.org` also can be in `source/url` (packages `flask-cors`, `phonenumbers`, etc.), and `pypi.python.org` (the `simplegeneric` package)

File: anaconda_linter/lint/check_build_help.py [hunk 2]
```diff
@@ -195,7 +195,157 @@ def check_source(self, source, section):
...
+        elif isinstance(source, list):
+            for num, src in enumerate(source):
+                pypi_source = "pypi.io" in source.get("url", "")
+                if pypi_source:
+                    break
```
Comment (skupr-anaconda) [inline anaconda_linter/lint/check_build_help.py:304]: `check_recipe` repeats three times. Does it make sense to create one function `check_pypi_source` outside the class?

File: anaconda_linter/lint/check_build_help.py (hunk 1 above)
Comment (cbouss) [reply anaconda_linter/lint/check_build_help.py]: addressed in new commit

File: anaconda_linter/lint/check_build_help.py (hunk 2 above)
Comment (cbouss) [reply anaconda_linter/lint/check_build_help.py]: addressed in new commit

File: anaconda_linter/lint/check_build_help.py [hunk 3]
```diff
@@ -195,7 +195,157 @@ def check_source(self, source, section):
...
+          - run_test.bat   # [win]
+
+    """
+
+    def check_recipe(self, recipe):
+        commands = recipe.get("test/commands", [])
+        if commands != []:
+            if set(os.listdir(recipe.recipe_dir)).intersection(
+                {"run_test.sh", "run_test.py", "run_test.bat"}
+            ):
```
[the remark this answers is not in the document]
Comment (cbouss) [reply anaconda_linter/lint/check_build_help.py]: addressed in new commit

**Review by marcoesters — APPROVED** [review]

**Pull request merged**